.github/workflows/build.yml

```diff
@@ -0,0 +1,89 @@
+name: Build wheels and deploy
+
+on:
+  workflow_dispatch:
+    inputs:
+      confirm_ref:
+        description: "Enter branch/tag name to deploy to PyPI, or leave blank to only build wheels."
+        default: ""
+      override_version:
+        description: "Override version number:"
+        default: ""
+
+
+jobs:
+  deploy_canary:
+    name: "Will deploy to PyPI if ticked"
+    runs-on: ubuntu-18.04
+    if: endsWith(github.ref, format('/{0}', github.event.inputs.confirm_ref))
+    steps:
+      - name: "Built wheels will be deployed at the end."
+        run: |
+          echo "Built wheels will be deployed at the end."
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-latest, macos-latest]
+    env:
+      # Set up wheels matrix.  This is CPython 3.6--3.9 for all OS targets.
+      CIBW_BUILD: "cp3[6-9]-*"
+      # Environment variable understood by QuTiP's setup.py to produce a release
+      # version of the wheels.
+      CIBW_ENVIRONMENT: "CI_QUTIP_RELEASE=1"
+      OVERRIDE_VERSION: ${{ github.event.inputs.override_version }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          # This is about the build environment, not the released wheel version.
+          python-version: '3.7'
+
+      - name: Install cibuildwheel
+        run: |
+          # cibuildwheel does the heavy lifting for us. Originally tested on
+          # 1.8.0, but should be fine at least up to any major new release.
+          python -m pip install 'cibuildwheel<2.0'
+
+      - name: Install Visual C++ for Python 2.7
+        if: runner.os == 'Windows'
+        run: |
+          choco install vcpython27 -f -y
```

Comment:
What do we need Python 2.7 for here?

Reply:
It's part of the cibuildwheel default action - I think it's because the lowest common denominator for Windows CI runners is assuming Python 2.7.  This is just bootstrapping cibuildwheel; once it starts running, I think it pulls in some Docker images that have a proper Python 3.x environment all ready to go, and does the build inside them.

Reply:
@hodgestar: on further thought, it's possible I'm wrong on this and it may only be needed if you want to build Python 2.7 wheels too.  I should check if it works without this line before I merge in unnecessary stuff.

Reply:
I tested it, and this isn't necessary for what we want to build.  I've removed it.